component/loki/source/journal/internal/target/journaltarget.go

```diff
@@ -211,12 +216,16 @@ func journalTargetWithReader(
 		for {
 			err := t.r.Follow(until, io.Discard)
 			if err != nil {
-				level.Error(t.logger).Log("msg", "received error during sdjournal follow", "err", err.Error())
+				if err == sdjournal.ErrExpired {
+					return
+				}
```

Comment:
nit: could we inline this along with the following errors? We'd also get the error message logged.